apps/antalmanac/src/components/Calendar/TermViewer.tsx

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useState } from 'react';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+
+import { ClassNameMap, Styles } from '@material-ui/core/styles/withStyles';
+import { Theme, withStyles } from '@material-ui/core/styles';
+import { termData } from '$lib/termData';
+import AppStore from '$stores/AppStore';
+
+const styles: Styles<Theme, object> = {
+    rootTermSelector: {
+        textAlign: 'center',
+    },
+};
+
+interface TermViewerProps {
+    classes: ClassNameMap;
+}
+
+const TermViewer = ({ classes }: TermViewerProps) => {
+    const getTerm = () => {
+        return AppStore.schedule.getCurrentScheduleTerm();
+    };
```

Comment:
This doesn't even have to be inside the component; this function can be moved outside and prevented from being re-calculated on every render.